File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String ans = strs[0];
        int n = strs.length;

        for(int i=1; i<n; i++) {
            String word = strs[i];
            int min = Math.min(ans.length(), word.length());
            int j;

            for(j=0; j<min; j++) {
                if(ans.charAt(j) != word.charAt(j)) break;
            }

            ans = ans.substring(0, j);
        }

        return ans;

    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size();
        string ans = strs[0];

        for(int i=1; i<n; i++) {
            int end = -1;
            for(int j=0; j<min(ans.length(), strs[i].length()); j++) {
                if(strs[i][j] != ans[j]) {
                    end = j;
                    break;
                }
            }
            if(end != -1) {
                ans = ans.substr(0, end);
            } else {
                if(strs[i].length() < ans.length()) ans = strs[i];
            }

            if(ans.length() == 0) return "";
        }
        return ans;
    }
};